packages/google_atlas/lib/src/google_atlas.dart

```diff
@@ -105,15 +119,36 @@ class _GoogleMapsProviderState extends State<GoogleMapsProvider> {
   }
 
   /// Converts an `Atlas.Marker` to a `GoogleMaps.Marker`
-  GoogleMaps.Marker _toGoogleMarker(Marker marker) {
-    return GoogleMaps.Marker(
-      markerId: GoogleMaps.MarkerId(marker.id),
-      position: GoogleMaps.LatLng(
-        marker.position.latitude,
-        marker.position.longitude,
-      ),
-      onTap: marker.onTap,
+  Future<Set<GoogleMaps.Marker>> _toGoogleMarkers(Set<Marker> markers) async {
+    Set<GoogleMaps.Marker> googleMarkers = Set();
+
+    for (Marker marker in markers) {
+      googleMarkers.add(GoogleMaps.Marker(
+        markerId: GoogleMaps.MarkerId(marker.id),
+        position: GoogleMaps.LatLng(
+          marker.position.latitude,
+          marker.position.longitude,
+        ),
+        onTap: marker.onTap,
+        icon:
+            marker.icon == null ? null : await _toBitmapDescriptor(marker.icon),
+      ));
+    }
+
+    return googleMarkers;
+  }
+
+  /// Converts `Atlas.MapIcon` to `GoogleMaps.BitmapDescriptor`
+  Future<GoogleMaps.BitmapDescriptor> _toBitmapDescriptor(
+      MapIcon mapIcon) async {
+    ImageConfiguration imageConfiguration = ImageConfiguration(
```

Comment:
Mark this variable as final (`final imageConfiguration = ...`), since its value is not mutated, and the compiler will infer the type